src/cluster/placement/types.go

```diff
@@ -445,6 +446,35 @@ type Options interface {
 	SetNowFn(fn clock.NowFn) Options
 }
 
+// SelectorOptions controls which instances are selected to be added into the placement.
+type SelectorOptions interface {
+	// AllowAllZones will enable the placement to contain hosts that
+	// are not contained within the same zone of the actual placement. This is
+	// needed for services that require cross zone communication.
+	AllowAllZones() bool
+	SetAllowAllZones(allowAllZones bool) SelectorOptions
+
+	// ValidZone is the zone that added instances must be in in order to
+	// be added to a placement. By default the valid zone will be the zone of
+	// instances already in a placement, however if a placement is empty then
+	// it is necessary to specify the valid zone when adding the first
+	// instance.
+	ValidZone() string
+
+	// SetValidZone is the zone that added instances must be in. See ValidZone.
+	SetValidZone(z string) SelectorOptions
+
+	// AddAllCandidates determines whether the placement will attempt to add all
+	// candidates when adding instances or just a single one.
+	AddAllCandidates() bool
+
+	// SetAddAllCandidates sets AddAllCandidates.
+	SetAddAllCandidates(addAllCandidates bool) SelectorOptions
+
+	InstrumentOptions() instrument.Options
+	SetInstrumentOptions(opts instrument.Options) SelectorOptions
```

Comment:
here too

Reply:
Huh weird that lint didn't get that, will add.